tests/unit/plugins/inventory/test_digitalocean.py

```diff
@@ -0,0 +1,178 @@
+# Copyright (c) 2021 Ansible Project
+# GNGeneral Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import pytest
+
+from ansible.errors import AnsibleParserError
+from ansible.inventory.data import InventoryData
+from ansible_collections.community.digitalocean.plugins.inventory.digitalocean import InventoryModule
+
+
+@pytest.fixture(scope="module")
+def inventory():
+    r = InventoryModule()
+    r.inventory = InventoryData()
+    return r
+
+
+def test_verify_file_bad_config(inventory):
+    assert inventory.verify_file('digitalocean_foobar.yml') is False
+
+
+def test_validate_config(inventory):
+    config = {
+        'plugin': 'community.digitalocean.digitalocean',
+        'api_token': 'xxxxxxxxxxxxxxxxxxxx',
+        'attributes': ['id', 'locked', 'foo', 'size'],
+        'var_prefix': 'foo_',
+        'pagination': 25
+    }
+    with pytest.raises(AnsibleParserError) as error_message:
+        inventory._validate_config(config)
+        assert "invalid attribute: foo" in error_message
```

Comment:
Nitpick:

```suggestion
    with pytest.raises(AnsibleParserError, match='invalid attribute: foo'):
        inventory._validate_config(config)
```

Reply:
Alternatively de-dent the `assert`. The current way, the assertion won't be checked.

Reply:
what to assert if no error message? 😕

Reply:
it has been deleted